The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/11654952772/job/32449063833) [(plain)](https://github.com/rust-lang/rust/commit/ca1d0db042fe826650128fe3d466d999cd533abe/checks/32449063833/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---
#13 2.957 Building wheels for collected packages: reuse
#13 2.958   Building wheel for reuse (pyproject.toml): started
#13 3.202   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 3.203   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132720 sha256=026f3bb0f1aa8090b861fd0a0939cb1a782396d84c8aab7875096557d637a0f6
#13 3.203   Stored in directory: /tmp/pip-ephem-wheel-cache-ty7ukqix/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#13 3.206 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#13 3.592 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#13 3.592 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#13 4.113 Collecting virtualenv
#13 4.113 Collecting virtualenv
#13 4.201   Downloading virtualenv-20.27.1-py3-none-any.whl (3.1 MB)
#13 4.332      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.1/3.1 MB 24.3 MB/s eta 0:00:00
#13 4.385 Collecting platformdirs<5,>=3.9.1
#13 4.406   Downloading platformdirs-4.3.6-py3-none-any.whl (18 kB)
#13 4.442 Collecting filelock<4,>=3.12.2
#13 4.481 Collecting distlib<1,>=0.3.7
#13 4.500   Downloading distlib-0.3.9-py2.py3-none-any.whl (468 kB)
#13 4.508      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 469.0/469.0 KB 84.5 MB/s eta 0:00:00
#13 4.508      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 469.0/469.0 KB 84.5 MB/s eta 0:00:00
#13 4.586 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 4.757 Successfully installed distlib-0.3.9 filelock-3.16.1 platformdirs-4.3.6 virtualenv-20.27.1
#13 DONE 4.8s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      169920 kB
DirectMap2M:     8218624 kB
DirectMap1G:    10485760 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
    Finished `dev` profile [unoptimized] target(s) in 0.04s
##[endgroup]
WARNING: `rust.download-rustc` is enabled. The `rust.channel` option will be overridden by the CI rustc's channel.
downloading https://static.rust-lang.org/dist/2024-10-16/rustfmt-nightly-x86_64-unknown-linux-gnu.tar.xz
---
fmt check
fmt: checked 5653 files
tidy check
tidy: Skipping binary file check, read-only filesystem
tidy error: /checkout/src/tools/rustc-std-workspace/rustc-std-workspace-core/Cargo.toml doesn't have `edition = "2021"` or `edition = "2024"` on a separate line
tidy error: /checkout/src/tools/rustc-std-workspace/rustc-std-workspace-std/Cargo.toml doesn't have `edition = "2021"` or `edition = "2024"` on a separate line
tidy error: /checkout/src/tools/rustc-std-workspace/rustc-std-workspace-alloc/Cargo.toml doesn't have `edition = "2021"` or `edition = "2024"` on a separate line
removing old virtual environment
creating virtual environment at '/checkout/obj/build/venv' using 'python3.10'
Requirement already satisfied: pip in ./build/venv/lib/python3.10/site-packages (24.3.1)
All checks passed!
checking C++ file formatting
some tidy checks failed
some tidy checks failed
Command has failed. Rerun with -v to see more details.
  local time: Sun Nov  3 21:59:32 UTC 2024
  network time: Sun, 03 Nov 2024 21:59:33 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -3,6 +3,9 @@ resolver = "2"
 members = [
   "compiler/rustc",
   "src/etc/test-float-parse",
+  "src/rustc-std-workspace/rustc-std-workspace-core",
+  "src/rustc-std-workspace/rustc-std-workspace-alloc",
+  "src/rustc-std-workspace/rustc-std-workspace-std",
   "src/rustdoc-json-types",
   "src/tools/build_helper",
   "src/tools/cargotest",

--- a/library/rustc-std-workspace-core/README.md
+++ b/library/rustc-std-workspace-core/README.md
@@ -27,3 +27,6 @@ it'll look like
 
 when Cargo invokes the compiler, satisfying the implicit `extern crate core`
 directive injected by the compiler.
+
+The sources for the crates.io version can be found in
+[`src/rustc-std-workspace`](../../src/rustc-std-workspace).

--- a/src/rustc-std-workspace/README.md
+++ b/src/rustc-std-workspace/README.md
@@ -1,4 +1,4 @@
-See `library/rustc-std-workspace-core/README.md` for context.
+See [`library/rustc-std-workspace-core/README.md`](../../library/rustc-std-workspace-core/README.md) for context.
 
 These are the crates.io versions of these crates, as opposed to the versions
 in `library` which are the ones used inside the rustc workspace.

--- a/src/rustc-std-workspace/rustc-std-workspace-alloc/Cargo.toml
+++ b/src/rustc-std-workspace/rustc-std-workspace-alloc/Cargo.toml
@@ -1,9 +1,11 @@
 [package]
 name = "rustc-std-workspace-alloc"
-version = "1.0.0"
+version = "1.0.1"
 authors = ["Alex Crichton <alex@alexcrichton.com>"]
-edition = "2018"
+edition = "2021"
 license = 'MIT/Apache-2.0'
-description = 'workspace hack'
+description = """
+crate for integration of crates.io crates into rust-lang/rust standard library workspace
+"""
 
-[dependencies]
+repository = "https://github.com/rust-lang/rust/tree/master/src/rustc-std-workspace"

--- a/src/rustc-std-workspace/rustc-std-workspace-alloc/src/lib.rs
+++ b/src/rustc-std-workspace/rustc-std-workspace-alloc/src/lib.rs
@@ -1,0 +1,3 @@
+#![no_std]
+extern crate alloc as the_alloc;
+pub use the_alloc::*;

--- a/src/rustc-std-workspace/rustc-std-workspace-core/Cargo.toml
+++ b/src/rustc-std-workspace/rustc-std-workspace-core/Cargo.toml
@@ -1,10 +1,11 @@
 [package]
 name = "rustc-std-workspace-core"
-version = "1.0.0"
+version = "1.0.1"
 authors = ["Alex Crichton <alex@alexcrichton.com>"]
+edition = "2021"
 license = "MIT/Apache-2.0"
 description = """
-Explicitly empty crate for rust-lang/rust integration
+crate for integration of crates.io crates into rust-lang/rust standard library workspace
 """
 
-[dependencies]
+repository = "https://github.com/rust-lang/rust/tree/master/src/rustc-std-workspace"

--- a/src/rustc-std-workspace/rustc-std-workspace-core/src/lib.rs
+++ b/src/rustc-std-workspace/rustc-std-workspace-core/src/lib.rs
@@ -1,0 +1,3 @@
+#![no_std]
+extern crate core as the_core;
+pub use the_core::*;

--- a/src/rustc-std-workspace/rustc-std-workspace-std/Cargo.toml
+++ b/src/rustc-std-workspace/rustc-std-workspace-std/Cargo.toml
@@ -2,9 +2,10 @@
 name = "rustc-std-workspace-std"
 version = "1.0.1"
 authors = ["Alex Crichton <alex@alexcrichton.com>"]
+edition = "2021"
 license = "MIT/Apache-2.0"
-description = "Workaround for rustbuild"
+description = """
+crate for integration of crates.io crates into rust-lang/rust standard library workspace
+"""
 
-[lib]
-name = "std"
-
+repository = "https://github.com/rust-lang/rust/tree/master/src/rustc-std-workspace"

